parlai/core/worlds.py

```diff
@@ -550,6 +550,13 @@ def __next__(self):
 
     def batch_observe(self, index, batch_actions, index_acting):
         batch_observations = []
+        a = self.world.get_agents()[index]
+
+        if (batch_actions is not None and len(batch_actions) > 0 and
+                hasattr(a, 'batch_observe')) and index != index_acting:
+            # if there's anything that needs to be batched, do it here
+            a.batch_observe(batch_actions)
```

Comment:
i don't think this will work in the case of hasattr(w, 'observe'), in that case the agents are not supposed to see the batch actions, only observe through the world in that case...
maybe it's better to put it on line 574.. after the batch_observations have been built??
would need a comment explaining. i think we should get alex and adam's view too